index.js

```diff
@@ -394,20 +394,20 @@ function inject (bot) {
         bot.setControlState('jump', true)
         canPlace = placingBlock.y + 1 < bot.entity.position.y
       }
+      bot.equip(block, 'hand', () => {})
```

Comment:
equip is async, if you are not using the callback you should await the promise. This change is likely to introduce more bug than it fix.

The problem you are trying to fix is a race condition when the tick loop is called again before the equip is done. The correct way to solve that would be to add a "lock" ie a variable that is checked first when entering the loop and set to true before doing any async action and false after.

```js
if (equipLock) return // we are already equiping something
equipLock = true // take the lock
await bot.equip(block, 'hand')
equipLock = false // done, release the lock
```

An alternative way of solving this is to cancel the first equip if its called again (we do that for a few other functions like dig or consume). But that would require modifications to equip directly.

@rom1504 any opinion ?

Reply:
Yes my idea was that it continues when it detects that the async equip finishes. I think the problem is that the async functions are used at all or at least how they are used at the moment. 
When I was testing I also noticed that the bot might try and place multiple blocks in the same position as it calls the async blockPlacement function whenever monitorMovement is called.
There should maybe be a lock system that waits for async actions to finish. But you have to make it reliable and resetable in case the path gets recalculated.

Reply:
It might be a good idea to think about splitting index.js into smaller files. If index.js is the package entry file but also handles all movement it can get a bit confusing.